core/state/state_object.go

```diff
@@ -87,23 +87,29 @@ type StateObject struct {
 	deleted bool
 }
 
-// accountData is the Ethereum consensus representation of accounts.
+// Account is the Ethereum consensus representation of accounts.
 // These objects are stored in the main account trie.
-type accountData struct {
+type Account struct {
 	Nonce    uint64
 	Balance  *big.Int
 	Root     common.Hash // merkle root of the storage trie
 	CodeHash []byte
 }
 
-func NewStateObject(address common.Address, db trie.Database) *StateObject {
-	object := &StateObject{
-		address: address,
-		data:    accountData{Balance: new(big.Int), CodeHash: emptyCodeHash},
-		storage: make(Storage),
-		dirty:   true,
+// NewObject creates a state object.
+func NewObject(address common.Address, data Account) *StateObject {
+	if data.Balance == nil {
+		data.Balance = new(big.Int)
```

Comment:
Mutating one of the passed in objects in a constructor seems like a Bad Thing to be doing.